lib/imports/validator.js

```diff
@@ -0,0 +1,40 @@
+import validator from 'jsen'
+import fieldValidation from './field-validation'
+
+export function validateDocField({ doc, field, newValue }) {
+  var data = doc.content || doc.facts
+  if (!data) throw new Meteor.Error('No valid data in document')
+  var docSchemaKey = doc.schema[field]
+  var schemaVal = {
+    type: 'string',
+    required: true
+  }
+  if (docSchemaKey) {
+    Object.assign(schemaVal, fieldValidation[docSchemaKey.type] || {})
+    Object.assign(schemaVal, docSchemaKey.validation || {})
+  }
+  var miniSchema = {
+    required: true,
+    type: 'object',
+    properties: { [field]: schemaVal },
+    additionalProperties: false
+  }
+  var validate = validator(miniSchema)
+  var valid = validate({ [field]: newValue })
+  if (!valid) return {
+    message: validationError(validate.errors[0].keyword)
+  }
+}
```

Comment:
Yes, that would work fine.  I did things this way so that any truthy value would indicate a validation error, rather than having to check the return value's keys, but I can understand it being confusing.